src/routes/process-user-vocabulary/components/AudioRecordOverlay.tsx

```diff
@@ -0,0 +1,205 @@
+import React, { Dispatch, ReactElement, SetStateAction, useEffect, useMemo, useState } from 'react'
+import { Modal as RNModal, Platform, Pressable } from 'react-native'
+import AudioRecorderPlayer from 'react-native-audio-recorder-player'
+import { PERMISSIONS } from 'react-native-permissions'
+import { heightPercentageToDP as hp } from 'react-native-responsive-screen'
+import uuid from 'react-native-uuid'
+import styled, { useTheme } from 'styled-components/native'
+
+import { CloseIcon, MicrophoneIcon } from '../../../../assets/images'
+import NotAuthorisedView from '../../../components/NotAuthorisedView'
+import { HeadingText } from '../../../components/text/Heading'
+import useGrantPermissions from '../../../hooks/useGrantPermissions'
+import { getLabels } from '../../../services/helpers'
+import { reportError } from '../../../services/sentry'
+
+const Container = styled.SafeAreaView`
+  flex: 1;
+  background-color: ${props => props.theme.colors.backgroundAccent};
+`
+
+const Icon = styled.Pressable`
+  align-self: flex-end;
+  margin: ${props => `${props.theme.spacings.xs} ${props.theme.spacings.md}`};
+`
+
+const RecordIcon = styled(Pressable)<{ isPressed: boolean }>`
+  width: ${hp('12%')}px;
+  height: ${hp('12%')}px;
+  align-self: center;
+  align-items: center;
+  justify-content: center;
+  border-radius: 100px;
+  background-color: ${props =>
+    props.isPressed ? props.theme.colors.audioRecordingActive : props.theme.colors.disabled};
+`
+
+const Content = styled.View`
+  margin-top: ${hp('12%')}px;
+  margin-bottom: ${hp('6%')}px;
+  flex: 1;
+  justify-content: space-between;
+`
+
+const Heading = styled(HeadingText)`
+  text-align: center;
+  font-size: ${props => props.theme.spacings.lg};
+`
+
+const HeadingContainer = styled.View`
+  min-height: ${hp('12%')}px;
+`
+
+const RecordingInfo = styled.Text`
+  font-size: ${props => props.theme.spacings.lg};
+  text-align: center;
+`
+
+const MeteringInfo = styled.View`
+  flex-direction: row;
+  align-self: center;
+  min-height: ${hp('12%')}px;
+`
+
+const MeteringBar = styled.View<{ height: number }>`
+  min-height: ${props => props.theme.spacings.xxs};
+  height: ${props => `${props.height * 2}`}px;
+  width: ${props => props.theme.spacings.xxs}
+  background-color: ${props => props.theme.colors.audioRecordingActive};
+  border-radius-: 50px;
+  align-self: center;
+  margin: 0 1px;
+`
+
+const InfoContainer = styled.View`
+  justify-content: flex-end;
+`
+
+interface AudioRecordOverlayProps {
+  onClose: () => void
+  setShowAudioRecordOverlay: Dispatch<SetStateAction<boolean>>
+  audioRecorderPlayer: AudioRecorderPlayer
+  setRecordingPath: Dispatch<SetStateAction<string | null>>
+  recordingPath: string | null
+  setRecordSeconds: Dispatch<SetStateAction<number>>
```

Comment:
@steffenkleinle  i don't understand this suggestion. These interface are correct

Reply:
Yeah, but I think they are too specific imo. AudioRecordOverylay should not care if it is a function updating a react state, a function updating redux state or a function doing something completely else. The additional specificity doesn't add any benefit and only may have to be changed in the future. For example if we at some point have to do something else when the recording seconds change instead of just updating the state we have to change the types here (may also already be the case depending on the other comments on not using useEffects on state changes).

Reply:
alright i got you point thx for explanation